proxywasm/proxytest/http_test.go

```diff
@@ -120,3 +121,30 @@ func TestBodyBuffering(t *testing.T) {
 		})
 	}
 }
+
+func TestProperties(t *testing.T) {
+	t.Run("Set and get properties", func(t *testing.T) {
+		host, reset := NewHostEmulator(NewEmulatorOption().WithVMContext(&testPlugin{}))
+		defer reset()
+
+		_ = host.InitializeHttpContext()
+
+		propertyPath := []string{
+			"route_metadata",
+			"filter_metadata",
+			"envoy.filters.http.wasm",
+			"hello",
+		}
+		propertyData := []byte("world")
+
+		err := host.SetProperty(propertyPath, propertyData)
+		require.Equal(t, err, nil)
+
+		data, err := host.GetProperty(propertyPath)
```

Comment:
This tests a round-trip for the host itself, but doesn't actually test the plugin reading / setting properties. I think we need a separate test plugin that reads, appends, and sets a property and then we'd have a full test.

Reply:
I've added an example and e2e test. Please, let me know what you think.